### ISSUE TITLE
ScanBatcher can create LidarScans with headers from old frames

### ISSUE THREAD

**Issue by jsundh** [ouster-lidar/ouster_example#300]
TL;DR: The LidarScan headers must be zeroed out in ScanBatcher along with the data in case of missing columns (at least from the beginning).

We recently made a recording (of the packet topics through ROS) where we saw a strange issue: Some PointCloud2 messages would occasionally get the wrong timestamp. More specifically, the exact same timestamp as the message two messages prior. 

Further investigation revealed that we had packet drops at these occasions, where  the beginning of the frame with the wrong timestamp was dropped. Extracting the Frame and Measurement IDs from the measurement blocks in the packets around one such drop showed:

| Frame ID | First Measurement ID | Last Measurement ID | Cloud message ts (nsec part) |
|---------:|---------------------:|--------------------:|-----------------------:|
|     7954 |                    0 |                1023 |         **.796607232** |
|     7955 |                    0 |             **911** |             .846782720 |
|     7956 |              **288** |                1023 |         **.796607232** |
|     7957 |                    0 |                1023 |             .947003648 |

Because the LidarScan `ls` in `os_cloud_node.cpp` and `ls_write` in `ScanBatcher` [get swapped](https://github.com/ouster-lidar/ouster_example/blob/47f25ed29ab0b4b6d32c3209fa1b53ea46751d0c/ouster_client/src/lidar_scan.cpp#L111) for every new frame, when frame 7956 is finalized and swapped from `ls_write` to `ls` to be published, the LidarScan instance still contains the first 288 headers from the complete frame 7954 two swaps earlier. The [`lidar_handler`](https://github.com/ouster-lidar/ouster_example/blob/47f25ed29ab0b4b6d32c3209fa1b53ea46751d0c/ouster_ros/src/os_cloud_node.cpp#L66) then uses the timestamp from the first measurement block header since it is non-zero.

The fix seems pretty straight forward: zero out the missing headers along with the data. I will submit a pull request shortly.

This bug might also affect other parts that use the `ScanBatcher`, but I have not looked into that. The same fix should apply to all cases though since it just changes how `ScanBatcher` works.

**Comment by dgarbuzov** [comment]
Great find, think this bug has existed for a long time. We also ran into it while adding dual returns support, so it should be addressed in (#313). Added a unit test for this in `test_batching.py`. Please reopen if anything still looks off.
